File: apps/web/src/types/customer.ts
export interface Customer {
  id: string;
  first_name: string;
  last_name: string;
  full_name: string;
  phone: string | null;
  whatsapp: string | null;
  email: string | null;
  address: string | null;
  document_id: string | null;
  active: boolean;
  created_at: string;
}

export interface CreateCustomerRequest {
  firstName: string;
  lastName: string;
  phone?: string;
  whatsapp?: string;
  email?: string;
  address?: string;
  documentId?: string;
}

export interface CustomersPage {
  data: Customer[];
  meta: { total: number; page: number; limit: number };
}

File: apps/api/src/main/java/rd/tallerfacil/api/customer/domain/Customer.java
package rd.tallerfacil.api.customer.domain;

import jakarta.persistence.*;
import lombok.*;
import rd.tallerfacil.api.shared.domain.TenantAwareEntity;

@Entity
@Table(name = "customers")
@Getter
@Setter
@NoArgsConstructor
public class Customer extends TenantAwareEntity {

    @Column(name = "first_name", nullable = false, length = 100)
    private String firstName;

    @Column(name = "last_name", nullable = false, length = 100)
    private String lastName;

    @Column(length = 20)
    private String phone;

    @Column(length = 20)
    private String whatsapp;

    @Column(length = 255)
    private String email;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Column(name = "document_id", length = 20)
    private String documentId;

    @Column(nullable = false)
    private boolean active = true;
}
